File: zcgl/src/main/java/com/bjprd/zcgl/utils/SPUtils.java
package com.bjprd.zcgl.utils;


import com.bjprd.zcgl.App;

/**
 * Created by 王少岩 on 2016/8/22.
 */
public class SPUtils {
    public static final String kUser_login = "login_flag";
    public static final String kUser_token = "token";
    public static final String kUser_account = "account";
    public static final String kUser_password = "password";
    public static final String kUser_remember = "remember";

    public static boolean isLogin() {
        return App.getAppContext().getUserPreference().getBoolean(kUser_login, false);
    }

    public static void setLoginStatus(boolean isLogin){
        App.getAppContext().getUserPreference().edit().putBoolean(kUser_login, isLogin).commit();
    }

    public static String getUserToken() {
        return App.getAppContext().getUserPreference().getString(kUser_token, "");
    }

    public static void setUserToken(String userToken) {
        App.getAppContext().getUserPreference().edit().putString(kUser_token, userToken).commit();
    }

    public static void setUserAccount(String account){
        App.getAppContext().getUserPreference().edit().putString(kUser_account, account).commit();
    }

    public static String getUserAccount(){
        return App.getAppContext().getUserPreference().getString(kUser_account, "");
    }

    public static void setUserPassword(String password){
        App.getAppContext().getUserPreference().edit().putString(kUser_password, password).commit();
    }

    public static String getUserPassword(){
        return App.getAppContext().getUserPreference().getString(kUser_password, "");
    }

    public static boolean isRemember() {
        return App.getAppContext().getUserPreference().getBoolean(kUser_remember, false);
    }

    public static void setRemember(boolean remember){
        App.getAppContext().getUserPreference().edit().putBoolean(kUser_remember, remember).commit();
    }

    /**
     * 用户登录，保持必要的数据
     */
    public static void onLogin(String token, String account, String password, boolean remember) {
        setUserToken(token);
        setUserAccount(account);
        setUserPassword(password);
        setRemember(remember);
        setLoginStatus(true);
    }
}

File: Zcgl/src/main/java/com/bjprd/zcgl/utils/SPUtils.kt
package com.bjprd.zcgl.utils


import com.bjprd.zcgl.App

/**
 * Created by 王少岩 on 2016/8/22.
 */
object SPUtils {
    val kUser_login = "login_flag"
    val kUser_token = "token"
    val kUser_account = "account"
    val kUser_password = "password"
    val kUser_remember = "remember"

    /**
     * 用户登录，保持必要的数据
     */
    fun onLogin(token: String, account: String, password: String, remember: Boolean) {
        save(kUser_token, token)
        save(kUser_account, account)
        save(kUser_password, password)
        save(kUser_remember, remember)
        save(kUser_login, true)
    }

    fun save(key: String, value: Any) {
        val editor = App.appContext!!.userPreference!!.edit()
        when (value) {
            is Int -> editor.putInt(key, value)
            is Float -> editor.putFloat(key, value)
            is Long -> editor.putLong(key, value)
            is Boolean -> editor.putBoolean(key, value)
            is String -> editor.putString(key, value)
            else -> throw Exception("not supported type")
        }
        editor.apply()
    }

    fun getString(key: String) :String = App.appContext!!.userPreference!!.getString(key, "")
    fun getInt(key: String) :Int = App.appContext!!.userPreference!!.getInt(key, 0)
    fun getBoolean(key: String) :Boolean = App.appContext!!.userPreference!!.getBoolean(key, false)
}
